### ISSUE TITLE
[IMPROVE] Matplotlib's Performance is Mind-Numbingly Slow. Simple Zooming and Panning Easily Overwhelms Modern Computers...

### ISSUE THREAD

**Issue by deeleeramone** [OpenBB-finance/OpenBBTerminal#2864]
The performance of Matplotlib is underwhelming at best; at worst is a complete deal-breaker. People will elect not to use this platform because the sloth-like performance of very basic charting functions provides a terrible and frustrating user experience with charts that don't have just basic cross-hair readouts. 

In a world full of interactive Notebooks, we have almost nothing to offer in terms of charting. This matters, a lot, and is a real problem that needs to be addressed.

Plotly provides a vastly superior charting experience for a Notebook, Dashboard, or Report. It actually works and is interactive. This matters, usability and functionality are in completely different leagues.  Even more importantly, it doesn't grind the entire system to a halt with inefficiencies.

**Comment by kairosdojo** [comment]
Not 100% sure this is the right place to discuss this but I feel your pain. Matplotlib is (unfortunately) a natural choice (actually: one of the few choices one has) in python-envs but unless you are generating static images it can only bring tears and pain. 

A couple of years back I have developed a system similar to Open BB, which I use daily, and back then I completely decoupled the computing / analysis / ML parts of it (in python) from the plotting feature - demanding it to Grafana. This means of course that all my data have to stay on an external DB / TSDB. Advantages: plots are updated in realtime, blazing fast, stable, accessible from wherever (on a webserver), you can annotate them, zero CPU sweating, etc.:

![FIaIrPEXIAEoS05](https://user-images.githubusercontent.com/22766463/196021542-1212a728-e3ea-42af-b872-5a53581b8fd1.png)

![FIPqp37WQAIEfJH](https://user-images.githubusercontent.com/22766463/196021543-481e74a9-5d28-4f60-9ac2-a984e8280091.png)

Obviously, though, this kind of approach would require an immense amount of code rewriting and architecture redesigning which I doubt might even be in plan right now.

My two cents.

**Comment by DidierRLopes** [comment]
TL;DR: This has been discussed multiple times, and is not a priority for us.

We have addressed this topic many times. OpenBB software is what it is today because it is:
* Open source
* Fully python based
* Free

The terminal is meant to be an all-in-one investment research platform, not only due to the data access it provides but due to the features that are built on top: econometrics, dashboards, reports, portfolio.

`matplotlib` is a python library that we decided to go very early on as it is a standard in the industry. At the end of the day this is only present on the `_view.py` files which render the data from the terminal, we could create `_view.py` that uses `plotly` instead but there is no ROI in that.

The reason I say there is no ROI is because we are about to announce OpenBB SDK which basically will allow developers to develop on top of our wrapper API using any plotting of library - including web frameworks. 

I will close this ticket.
